src/roswire/ros2/node.py

```diff
@@ -4,26 +4,50 @@
 import typing
 
 import attr
+import logger
+
+from ..interface import Node
+from .state import ROS2StateProbe
+import dockerblade
 
 if typing.TYPE_CHECKING:
     from ..app import AppInstance
 
 
 @attr.s(frozen=True, auto_attribs=True, slots=True)
-class ROS2Node:
+class ROS2Node(Node):
     """Provides an interface for interacting with a ROS2 node.
 
     Attributes
     ----------
     name: str
         The name of the node.
     """
-    app_instance: 'AppInstance'
+    app_instance: 'AppInstance' = attr.ib()
+    node_name: str = attr.ib()
```

Comment:
We can just use `name` here rather than `node_name`. We don't need the extra `@property` :slightly_smiling_face: